test/python3/libdnf5/comps/test_group.py

```diff
@@ -27,11 +25,36 @@ class TestGroup(base_test_case.BaseTestCase):
     def test_group(self):
         self.add_repo_repomd("repomd-comps-core")
         q_core = libdnf5.comps.GroupQuery(self.base)
-        core = q_core.get()
+        _ = q_core.get()
 
     def test_group_query_without_setup(self):
         # Create a new Base object
         base = libdnf5.base.Base()
 
         # Try to create a group query without running base.setup()
         self.assertRaises(RuntimeError, libdnf5.comps.GroupQuery, base)
+
+    def test_group_install_remove(self):
+        # Check group is correctly registered after install
+        # and so it could be removed after that
+        self.add_repo_repomd("repomd-comps-core")
+
+        goal_install = libdnf5.base.Goal(self.base)
+        goal_install.add_install("@core")
+
+        transaction_install = goal_install.resolve()
+        self.assertEqual(transaction_install.get_problems(),
+                         libdnf5.base.GoalProblem_NO_PROBLEM)
+        transaction_install.run()
+        self.assertFalse(transaction_install.get_transaction_problems())
+
+        self.repo_sack.update_and_load_enabled_repos(True)
```

Comment:
This is a thing I was struggling with, because without reloading the sack, the `goal_remove` object then don't know about the group installed state. I was wondering whether we could update the objects state automatically as we use the same one `base` throughout the script?

Reply:
I agree, that is odd. We should either handle that or at least document it. If I understand correctly then basically running a transaction invalidates the sack? Or is it just for groups/modules?

Reply:
Yes, it's a general thing.

Reply:
There is a fresh [issue](https://github.com/rpm-software-management/dnf5/issues/412) related to this. If there is really no working way how to reset the sack I think it's a bug, because users cannot really do more transactions within one API script.

Reply:
So I would postpone this until the issue is resolved, because the current solution calling the `update_and_load_enabled_repos` twice is a bad way how to solve this and also probably unwanted - it should be either documented as you said, or fixed to be callable only once.